controllers/hvpa_controller_test.go

```diff
@@ -77,6 +79,9 @@ var _ = Describe("#TestReconcile", func() {
 						},
 					},
 				},
+				Status: appsv1.DeploymentStatus{
```

Comment:
we don't expect `status` to be filled when creating a fresh HVPA. Should we have that as input to test framework then?

Reply:
> we don't expect status to be filled when creating a fresh HVPA. 

That is right. `kube-apiserver` would ignore the status part. But in this integration test, the `Deployment` test is never going to be updated by itself (no deployment controller).

Hence, the explicit [status update](https://github.com/gardener/hvpa-controller/pull/35/files#diff-beb07c813c94dc78f2e1057a5b8f96a4R108). Now, I could have used a different `Deployment` struct for this but used the existing one. Should I use a different struct?

Reply:
right. This should be fine. :)